### PR TITLE
fix a bug in libwinpr module

### PR DESCRIPTION
**Pull request by seasar**
when enqueue a new entry, we should rearrange wrapped entries if we invoke realloc().
the judgment condition: queue->tail < old_capacity   is wrong, it should be: queue->tail <= queue->head

### CODE REVIEW

File: winpr/libwinpr/utils/collections/MessageQueue.c [hunk 1]
```diff
@@ -82,8 +82,9 @@ void MessageQueue_Dispatch(wMessageQueue* queue, wMessage* message)
 		queue->capacity = new_capacity;
 		queue->array = (wMessage*) realloc(queue->array, sizeof(wMessage) * queue->capacity);
 		ZeroMemory(&(queue->array[old_capacity]), old_capacity * sizeof(wMessage));
-
-		if (queue->tail < old_capacity)
+		
+		//rearrange wrapped entries
```
Comment (akallabeth) [inline winpr/libwinpr/utils/collections/MessageQueue.c]: Please avoid C++ comments.

File: winpr/libwinpr/utils/collections/Queue.c [hunk 2]
```diff
@@ -148,7 +148,8 @@ void Queue_Enqueue(wQueue* queue, void* obj)
 		queue->array = (void**) realloc(queue->array, sizeof(void*) * queue->capacity);
 		ZeroMemory(&(queue->array[old_capacity]), old_capacity * sizeof(void*));
 
-		if (queue->tail < old_capacity)
+		//rearrange wrapped entries
```
Comment (akallabeth) [inline winpr/libwinpr/utils/collections/Queue.c]: Please avoid C++ comments.

**Comment by akallabeth** [comment]
@freerdp-bot test
@seasar Could you please fix the comments and squash your changes into a single commit?

**Comment by bmiklautz** [comment]
@seasar very nice finding! While reviewing I saw that the memory zeroed with is also incorrect. We use old_capacity instead of (new_capacity - old_capacity). Maybe you can fix this in your PR as well if you are in mood (otherwise I'll do a separate pull).

**Pull request closed without merging**